pkgs/build-support/setup-hooks/multiple-outputs.sh

```diff
@@ -4,16 +4,30 @@ preFixupHooks+=(_multioutDocs)
 preFixupHooks+=(_multioutDevs)
 postFixupHooks+=(_multioutPropagateDev)
 
-# Assign the first string containing nonempty variable to the variable named $1
+# _assignFirst varName otherVarNames*
+#
+# Set the value of the variable named $varName to the first of otherVarNames
+# that refers to a non-empty variable name.
+#
+# If none of otherVarNames refers to a non-empty variable, the error message is
+# specific to this function's use case, which is setting up the output variables.
 _assignFirst() {
     local varName="$1"
     local REMOVE=REMOVE # slightly hacky - we allow REMOVE (i.e. not a variable name)
     shift
-    while (( $# )); do
-        if [ -n "${!1-}" ]; then eval "${varName}"="$1"; return; fi
-        shift
+    for var in "$@"; do
+        if [ -n "${!var-}" ]; then eval "${varName}"="${var}"; return; fi
     done
-    echo "Error: _assignFirst found no valid variant!"
+    echo
+    echo "error: _assignFirst: could not find a non-empty variable to assign to ${varName}. The following variables were all unset or empty: $*."
+    if [ -z "${out:-}" ]; then
+        echo '       If you do not want an "out" output in your derivation, make sure to define'
+        echo '       the other specific required outputs. This can be achieved by picking one'
+        echo "       of $* to add as an output."
```

Comment:
These needs some kind of highlighting, either with quotation marks or, better, by listing them in a line each.

Reply:
This is a rather niche error and, this being stdenv, it's not easy to test because of the rebuilds, so I'd rather spend time on something else.

Reply:
Of course. Not suggesting to do anything fancy. How about this? Just setting them off will make it easily readable without much fuzz.
```suggestion
        echo "       of the following to add as an output:"
        echo "           $*"
```

Reply:
I've committed the first suggestion. Repeating the variables on a separate line here seems to add too much weight, as they're a repetition of the variables above. I figured repeating them would be more readable than referencing them. It will like like `of bin dev to add as an output`. I don't think that's too bad when they were previously accentuated by having their own indented line.

Reply:
Right, then we can also just omit them entirely since they will be right in the previous line.

```suggestion
        echo "       of the above as an output."
```